File: src/main/scala/com/yhsif/autonotif/PkgAdapter.scala
package com.yhsif.autonotif

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import scala.collection.mutable.ListBuffer

class PkgAdapter(var list: ListBuffer[PkgData])
    extends RecyclerView.Adapter[PkgViewHolder] {

  override def onCreateViewHolder(parent: ViewGroup, vt: Int): PkgViewHolder = {
    val v =
      LayoutInflater
        .from(parent.getContext())
        .inflate(R.layout.pkg_item, parent, false)
    return new PkgViewHolder(v)
  }

  override def onBindViewHolder(vh: PkgViewHolder, i: Int): Unit = {
    vh.setIcon(list.apply(i).icon)
    vh.setName(list.apply(i).name)
    vh.setBackground(i)
  }

  override def getItemCount(): Int = list.length

  override def onAttachedToRecyclerView(rv: RecyclerView): Unit = {
    super.onAttachedToRecyclerView(rv)
  }

  def insert(i: Int, data: PkgData) = {
    list.insert(i, data)
    notifyItemInserted(i)
  }

  def remove(data: PkgData) = {
    val i = list.indexOf(data)
    if (i >= 0) {
      list.remove(i)
      notifyItemRemoved(i)
    }
  }
}

File: android/src/com/yhsif/notifbot/PkgAdapter.kt
package com.yhsif.notifbot

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class PkgAdapter(
    var list: MutableList<PkgData>,
    val listener: View.OnClickListener
) : RecyclerView.Adapter<PkgViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, vt: Int): PkgViewHolder {
        val v = LayoutInflater
            .from(parent.getContext())
            .inflate(R.layout.pkg_item, parent, false)
        v.setOnClickListener(listener)
        return PkgViewHolder(v)
    }

    override fun onBindViewHolder(vh: PkgViewHolder, i: Int) {
        vh.setIcon(list.get(i).icon)
        vh.setName(list.get(i).name)
        vh.setBackground(i)
    }

    override fun getItemCount(): Int = list.size

    override fun onAttachedToRecyclerView(rv: RecyclerView) {
        super.onAttachedToRecyclerView(rv)
    }

    fun remove(i: Int) {
        list.removeAt(i)
        notifyDataSetChanged()
    }
}
